### PR TITLE
#82 input.subject.groups to seal_subject.groups

### PR DESCRIPTION
**Pull request by abynenkov-ib**

### CODE REVIEW

File: pkg/compiler/rego/rego.go [hunk 1]
```diff
@@ -170,7 +170,7 @@ func (c *CompilerRego) compileSubject(sub ast.Subject) (string, error) {
 
 	switch t := sub.(type) {
 	case *ast.SubjectGroup:
-		return fmt.Sprintf("    seal_list_contains(input.subject.groups, `%s`)", t.Group), nil
+		return fmt.Sprintf("    seal_list_contains(seal_subject.groups, `%s`)", t.Group), nil
 	case *ast.SubjectUser:
 		return fmt.Sprintf("    input.subject.email == `%s`", t.User), nil
```
Comment (rchowinfoblox) [inline pkg/compiler/rego/rego.go:175]: also needs to change to "seal_subject". ie "input.subject" to "seal_subject".

File: docs/source/examples/petstore/petstore.all.rego [hunk 2]
```diff
@@ -61,44 +61,44 @@ line6_not1_cnd {
...
 
 allow {
-	seal_list_contains(input.subject.groups, `managers`)
+	seal_list_contains(seal_subject.groups, `managers`)
 	input.verb == `manage`
 	re_match(`petstore.*`, input.type)
 }
 
 allow {
-	input.subject.email == `cto@petstore.swagger.io`
```
Comment (abynenkov-ib) [inline docs/source/examples/petstore/petstore.all.rego:76]: @rchowinfoblox , @wplu - is it expected change?

Comment (rchowinfoblox) [reply docs/source/examples/petstore/petstore.all.rego:76]: I think if input jwt contains 'sub' claim, than 'seal_subject.sub' is correct.

Note that an Infoblox JWT doesn't seem to use 'sub' claim, but I guess that's outside the scope of issue.  Here's an example JWT from env-4:

`{
  "user_id": "77",
  "identity_user_id": "a5854665-feb5-4dce-9b9c-000000000077",
  "username": "akumar@infoblox.com",
  "account_id": "40",
  "csp_account_id": 40,
  "identity_account_id": "aa9b7c1d-9846-4106-97bb-000000000040",
  "account_name": "akumar",
  "account_domain": "infoblox.site",
  "account_number": "40",
  "company_number": 300040,
  "account_storage_id": 300040,
  "api_token": "1452bfd8a6a03a3bdac6a416dc86cda7",
  "sfdc_account_id": "199",
  "groups": [
    "ib-everyone-hidden",
    "act_admin"
  ],
  "subject": {
    "id": "akumar@infoblox.com",
    "subject_type": "user",
    "authentication_type": "token",
    "authentication_id": "747005d9-f829-417d-9550-000000001954"
  },
  "aud": "ib-ctk",
  "exp": 1603128285,
  "jti": "95ecc71d-c1e7-488b-9a78-c6b68f199d62",
  "iat": 1603127295,
  "iss": "identity",
  "nbf": 1603127295
}`

Comment (rchowinfoblox) [reply docs/source/examples/petstore/petstore.all.rego:76]: I've opened issue https://github.com/infobloxopen/seal/issues/94 regarding Infoblox JWT.

**Review by rchowinfoblox — APPROVED** [review]

File: docs/source/examples/petstore/petstore.all.rego (hunk 2 above)
Comment (wplu) [reply docs/source/examples/petstore/petstore.all.rego:76]: do not discuss infoblox specific things anymore

**Comment by abynenkov-ib** [comment]
moved to #95

**Pull request closed without merging**